include/swift/Serialization/ModuleFormat.h

```diff
@@ -344,6 +344,18 @@ enum SpecialModuleID : uint8_t {
   NUM_SPECIAL_MODULES
 };
 
+// Continuation of the special module IDs above for special names that don't
+// have an identifier. As above, these IDs must \em not be renumbered or
+// reordered without incrementing VERSION_MAJOR.
+enum SpecialNameID : uint8_t {
+  SUBSCRIPT_ID = NUM_SPECIAL_MODULES,
```

Comment:
This seems dangerous. How do people know whether to check NUM_SPECIAL_MODULES or NUM_SPECIAL_IDS? I think we should either fold the two lists together or allow the same value to represent multiple things (builtin module OR `subscript`, current module OR `init`, etc).